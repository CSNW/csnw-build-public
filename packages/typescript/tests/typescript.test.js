const { join } = require('path');
const { Files } = require('@csnw-build/pipeline');

const typescript = require('../')();
const cwd = join(__dirname, 'fixtures');

test('pipeline', async () => {
  const process = await typescript.pipeline({
    cwd,
    sourcemaps: true,
    cache: false
  });
  const files = await Files.from('*.{ts,tsx}', { cwd });

  const result = await process(files);
  expect(result).toMatchSnapshot();
});
