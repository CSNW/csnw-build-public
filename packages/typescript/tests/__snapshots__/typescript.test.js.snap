// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pipeline 1`] = `
Object {
  "a.js": Object {
    "baseDir": "C:/dev/csnw-build/packages/typescript/tests/fixtures",
    "contents": "\\"use strict\\";
Object.defineProperty(exports, \\"__esModule\\", { value: true });
var A = /** @class */ (function () {
    function A(value) {
        this.name = value;
    }
    return A;
}());
exports.A = A;
",
    "cwd": "C:/dev/csnw-build/packages/typescript/tests/fixtures",
    "path": "C:/dev/csnw-build/packages/typescript/tests/fixtures/a.js",
    "sourceMap": Object {
      "file": "a.js",
      "mappings": ";;AAAA;IAGE,WAAY,KAAa;QACvB,IAAI,CAAC,IAAI,GAAG,KAAK,CAAC;IACpB,CAAC;IACH,QAAC;AAAD,CAAC,AAND,IAMC;AANY,cAAC",
      "names": Array [],
      "sources": Array [
        "a.ts",
      ],
      "sourcesContent": Array [
        "export class A {
  name: string;

  constructor(value: string) {
    this.name = value;
  }
}
",
      ],
      "version": 3,
    },
  },
  "b.js": Object {
    "baseDir": "C:/dev/csnw-build/packages/typescript/tests/fixtures",
    "contents": "\\"use strict\\";
Object.defineProperty(exports, \\"__esModule\\", { value: true });
function Button(props) {
    return React.createElement(\\"button\\", null, props.children);
}
exports.Button = Button;
var B = /** @class */ (function () {
    function B() {
    }
    B.prototype.render = function () {
        return React.createElement(Button, null, \\"(message)\\");
    };
    return B;
}());
exports.B = B;
",
    "cwd": "C:/dev/csnw-build/packages/typescript/tests/fixtures",
    "path": "C:/dev/csnw-build/packages/typescript/tests/fixtures/b.js",
    "sourceMap": Object {
      "file": "b.js",
      "mappings": ";;AAAA,gBAAuB,KAAK;IAC1B,OAAO,oCAAS,KAAK,CAAC,QAAQ,CAAU,CAAC;AAC3C,CAAC;AAFD,wBAEC;AAED;IAAA;IAIA,CAAC;IAHC,kBAAM,GAAN;QACE,OAAO,oBAAC,MAAM,oBAAmB,CAAC;IACpC,CAAC;IACH,QAAC;AAAD,CAAC,AAJD,IAIC;AAJY,cAAC",
      "names": Array [],
      "sources": Array [
        "b.tsx",
      ],
      "sourcesContent": Array [
        "export function Button(props) {
  return <button>{props.children}</button>;
}

export class B {
  render() {
    return <Button>(message)</Button>;
  }
}
",
      ],
      "version": 3,
    },
  },
}
`;
